src/axom/core/Types.hpp

```diff
@@ -69,97 +68,128 @@ using IndexType = std::int32_t;
- * \see Types.cpp
- */
+// Note: MSVC complains about uninitialized static const integer class members,
+// but our nvcc/mpi combination complains about static constexpr MPI_Datatypes.
+// Since it's one line per traits class, implement both ways w/ an ifdef
+
+/// Traits class to map Axom types to their corresponding MPI type.
 template <class AxomType>
 struct mpi_traits
 {
+  #ifdef _MSC_VER
+  static constexpr MPI_Datatype type = MPI_DATATYPE_NULL;
+  #else
   static const MPI_Datatype type;
+  #endif
 };
 
 /// \name Specialization of mpi_traits
 /// @{
-
-//------------------------------------------------------------------------------
 template <>
 struct mpi_traits<float64>
 {
+  #ifdef _MSC_VER
+  static constexpr MPI_Datatype type = MPI_DOUBLE;
+  #else
   static const MPI_Datatype type;
+  #endif
 };
 
-//------------------------------------------------------------------------------
 template <>
 struct mpi_traits<float32>
 {
+  #ifdef _MSC_VER
+  static constexpr MPI_Datatype type = MPI_FLOAT;
+  #else
   static const MPI_Datatype type;
+  #endif
 };
 
-//------------------------------------------------------------------------------
 template <>
 struct mpi_traits<std::int8_t>
 {
+  #ifdef _MSC_VER
+  static constexpr MPI_Datatype type = MPI_INT8_T;
+  #else
   static const MPI_Datatype type;
+  #endif
 };
 
-//------------------------------------------------------------------------------
 template <>
 struct mpi_traits<std::uint8_t>
 {
+  #ifdef _MSC_VER
+  static constexpr MPI_Datatype type = MPI_UINT8_T;
+  #else
   static const MPI_Datatype type;
+  #endif
 };
 
-//------------------------------------------------------------------------------
 template <>
 struct mpi_traits<std::int16_t>
 {
+  #ifdef _MSC_VER
+  static constexpr MPI_Datatype type = MPI_INT16_T;
+  #else
   static const MPI_Datatype type;
+  #endif
 };
 
-//------------------------------------------------------------------------------
 template <>
 struct mpi_traits<std::uint16_t>
 {
+  #ifdef _MSC_VER
+  static constexpr MPI_Datatype type = MPI_UINT16_T;
+  #else
   static const MPI_Datatype type;
+  #endif
 };
 
-//------------------------------------------------------------------------------
 template <>
 struct mpi_traits<std::int32_t>
 {
+  #ifdef _MSC_VER
+  static constexpr MPI_Datatype type = MPI_INT32_T;
+  #else
   static const MPI_Datatype type;
+  #endif
 };
 
-//------------------------------------------------------------------------------
 template <>
 struct mpi_traits<std::uint32_t>
 {
+  #ifdef _MSC_VER
+  static constexpr MPI_Datatype type = MPI_UINT32_T;
+  #else
   static const MPI_Datatype type;
+  #endif
 };
 
-  //------------------------------------------------------------------------------
   #ifndef AXOM_NO_INT64_T
 template <>
 struct mpi_traits<std::int64_t>
 {
+    #ifdef _MSC_VER
+  static constexpr MPI_Datatype type = MPI_INT64_T;
+    #else
   static const MPI_Datatype type;
+    #endif
```

Comment:
```suggestion
  #ifdef _MSC_VER
  static constexpr MPI_Datatype type = MPI_INT64_T;
  #else
  static const MPI_Datatype type;
  #endif
```

Reply:
Unfortunately, the indentations are from clang-format.
We can disable formatting, or attempt to modify our `.clang-format` settings to achieve your recommendation (or live w/ the current indentation).

What's your preference?

Reply:
i say leave it as is. I expect we are going to be upgrading our clang-format soon so maybe it will be fixed?